frontend/src/js/controllers/analyticsCtrl.js

```diff
@@ -0,0 +1,296 @@
+// Invoking IIFE for dashboard
+(function() {
+
+    'use strict';
+
+    angular
+        .module('evalai')
+        .controller('AnalyticsCtrl', DashCtrl);
+
+    DashCtrl.$inject = ['utilities', '$state', '$rootScope'];
+
+    function DashCtrl(utilities, $state, $rootScope) {
+        var vm = this;
+
+        vm.hostTeam = {};
+        vm.teamId = null;
+        vm.currentTeamName = null;
+        vm.challengeListCount = 0;
+        vm.challengeList = {};
+        vm.challengeAnalysisDetail = {};
+        vm.isTeamSelected = false;
+        vm.challengeId = null;
+        vm.currentChallengeDetails = {};
+        vm.currentPhase = [];
+        vm.totalSubmission = [];
+        // vm.challengeCount = 0;
+        // vm.hostTeamCount = 0;
+        // vm.hostTeamExist = false;
+        // vm.participatedTeamCount = 0;
+        // // get token
+
+        // if (vm.teamId == null) {
+        //     vm.isTeamSelected = false;
+        // } else {
+        //     vm.isTeamSelected = true;
+        // }
+        var userKey = utilities.getData('userKey');
+
+        var parameters = {};
+        parameters.url = 'hosts/challenge_host_team/';
+        parameters.method = 'GET';
+        parameters.token = userKey;
+        parameters.callback = {
+            onSuccess: function(response) {
+                var status = response.status;
+                var details = response.data;
+                if (status == 200) {
+                    vm.hostTeam = details.results;
+
+                }
+            },
+            onError: function(response) {
+                var status = response.status;
+                var error = response.data;
+                if (status == 403) {
+                    vm.error = error;
+
+                    // navigate to permissions denied page
+                    $state.go('web.permission-denied');
+                } else if (status == 401) {
+                    alert("Timeout, Please login again to continue!");
+                    utilities.resetStorage();
+                    $state.go("auth.login");
+                    $rootScope.isAuth = false;
+
+                }
+            }
+        };
+        utilities.sendRequest(parameters);
+
+
+        parameters.url = 'challenges/challenge?mode=host';
```

Comment:
Writing same code three times to make requests for three endpoints does not seem a good solution.
## Proposed Approach
- Make a service name AJAX Requests
- Call this service whenever you want the data with parameters.